src/RA_Dlg_Memory.cpp

```diff
@@ -782,661 +783,662 @@ void Dlg_Memory::ClearLogOutput() noexcept
-                                _stprintf_s(buffer, sizeof(buffer), _T("%s"), _T("Invalid Range"));
-                            else
-                                _stprintf_s(buffer, sizeof(buffer), _T("%s"), NativeStr(sFirstLine).c_str());
-                        }
+                        const std::string& sFirstLine = m_SearchResults[m_nPage].m_results.Summary();
+                        if (sFirstLine.empty())
+                            _stprintf_s(buffer, sizeof(buffer), _T("%s"), _T("Invalid Range"));
                         else
-                        {
-                            SetTextColor(pDIS->hDC, RGB(0, 100, 150));
-                            const unsigned int nMatches = m_SearchResults[m_nPage].m_results.MatchingAddressCount();
-                            if (nMatches > MIN_RESULTS_TO_DUMP)
-                                _stprintf_s(buffer, sizeof(buffer),
-                                            _T("Found %u matches! (Displaying first %u results)"), nMatches,
-                                            MIN_RESULTS_TO_DUMP);
-                            else if (nMatches == 0)
-                                _stprintf_s(buffer, sizeof(buffer), _T("Found *ZERO* matches!"));
-                            else
-                                _stprintf_s(buffer, sizeof(buffer), _T("Found %u matches!"), nMatches);
-                        }
-
-                        DrawText(pDIS->hDC, buffer, _tcslen(buffer), &pDIS->rcItem,
-                                 DT_SINGLELINE | DT_LEFT | DT_NOPREFIX | DT_NOCLIP | DT_VCENTER | DT_END_ELLIPSIS);
-                        SetTextColor(pDIS->hDC, GetSysColor(COLOR_WINDOWTEXT));
+                            _stprintf_s(buffer, sizeof(buffer), _T("%s"), NativeStr(sFirstLine).c_str());
                     }
                     else
                     {
-                        auto& currentSearch = m_SearchResults[m_nPage];
-                        ra::services::SearchResults::Result result;
-                        if (!currentSearch.m_results.GetMatchingAddress(pDIS->itemID - 2, result))
-                            break;
+                        SetTextColor(lpDrawItem->hDC, RGB(0, 100, 150));
+                        const unsigned int nMatches = m_SearchResults[m_nPage].m_results.MatchingAddressCount();
+                        if (nMatches > MIN_RESULTS_TO_DUMP)
+                            _stprintf_s(buffer, sizeof(buffer),
+                                _T("Found %u matches! (Displaying first %u results)"), nMatches,
+                                MIN_RESULTS_TO_DUMP);
+                        else if (nMatches == 0)
+                            _stprintf_s(buffer, sizeof(buffer), _T("Found *ZERO* matches!"));
+                        else
+                            _stprintf_s(buffer, sizeof(buffer), _T("Found %u matches!"), nMatches);
+                    }
+                    auto myRect{lpDrawItem->rcItem};
```

Comment:
The Windows API passes` lpDrawItem` as a pointer to const, so we needed to copy it. `RECT` is a literal type so it shouldn't be expensive.